.github/workflows/grpc-go-check.yml

```diff
@@ -0,0 +1,33 @@
+name: Check generated go files
+
+on: push
+
+jobs:
+  check:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Install protoc
+        run: sudo apt-get update && sudo apt-get install -y protobuf-compiler && protoc --version
+
+      - name: Check gRPC generated files not manually changed
+        shell: bash
+        run: |
+          make generate-grpc-files
+          CHANGED=$(git diff --name-only ./**/*.go | wc -l)
+          if [ "${CHANGED}" -ne "0" ]; then
+            git diff ./**/*.go
+            echo "Committed Protobuf files do not match the newly generated. Please, regenerate Protobuf and commit changes"
+            exit 1
+          fi
```

Comment:
note: I did not know about the [--exit-code](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-code--exit-codecode) when I first wrote the script. Maybe we can migrate to using these options instead of chaining results with `wc -l`

Also, maybe, it is worth moving the check into the Makefile